language-server/src/jsonc-instance.js

```diff
@@ -181,6 +181,12 @@ export class JsoncInstance {
   textLength() {
     return this.node.length;
   }
+
+  findProperty(instance) {
+    const keyNode = findNodeAtOffset(instance.root, instance.node.children[0]?.children[0]?.offset);
+    const valNode = findNodeAtOffset(instance.root, instance.node.children[0]?.children[1]?.offset);
+    return { key: keyNode, value: valNode };
+  }
```

Comment:
This doesn't make sense. You create an instance and then pass it to itself? And at no point do you actually pass the position from the onCompletion event to this method. You need to use that position to find the offset (hint: `this.textDocument.???`). Then you can use that offset to get the node with `findNodeAtOffset`. Then use that node to create and return a JsoncInstance representing that position. To build the JsoncInstance, you need to build the JSON Pointer for that location as well (hint: `getNodePath` from jsonc-parser). You don't need to return the key because you'll have all the information you need from the pointer in the JsoncInstance.

Given that my suggestion changes what this method does and its inputs and outputs, it's going to need a better name that accurately describes what it does.